spec/Appendix B -- Grammar Summary.md

```diff
@@ -41,24 +47,41 @@ Token ::
 
 Punctuator :: one of ! $ & ( ) ... : = @ [ ] { | }
 
-Name :: /[_A-Za-z][_0-9A-Za-z]*/
+Name ::
+  - NameStart NameContinue* [lookahead != NameContinue]
+
+NameStart ::
+  - Letter
+  - `_`
+
+NameContinue ::
+  - Letter
+  - Digit
+  - `_`
 
-IntValue :: IntegerPart
+Letter :: one of
+  `A` `B` `C` `D` `E` `F` `G` `H` `I` `J` `K` `L` `M`
+  `N` `O` `P` `Q` `R` `S` `T` `U` `V` `W` `X` `Y` `Z`
+  `a` `b` `c` `d` `e` `f` `g` `h` `i` `j` `k` `l` `m`
+  `n` `o` `p` `q` `r` `s` `t` `u` `v` `w` `x` `y` `z`
```

Comment:
Why does Letter just have to be in this subset of characters? What about non-English languages?

Reply:
since name tokens are restricted to ASCII, this is done to be compatible with as many languages as possible since name token can represent type names, field names, enum values etc.

There are a lot of languages supporting UTF-8 or UTF-16 names but not all. So, with this restriction GraphQL can be easily translated to any programming language.

But this limitation does only apply to name tokens ... string tokens on the other hand allow for UTF-8.

And the case here is that if you have two tokens without a separator ... like `123abc`. So, with the new spec a float and int token are not allowed to be followed by any letter that can start a name token.

Reply:
That's right:

https://graphql.github.io/graphql-spec/draft/#sel-FAFJXJCAACCgDiyZ

Of course, in the future we could propose to expand this, but that's the current rule and rationale